src/main/java/io/anserini/collection/CovidCollection.java

```diff
@@ -0,0 +1,149 @@
+import java.nio.file.Paths;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+/**
+ * A document collection for the CORD-19 dataset provided by Semantic Scholar.
+ */
+public class CovidCollection extends DocumentCollection<CovidCollection.Document> {
+  private static final Logger LOG = LogManager.getLogger(CovidCollection.class);
+
+  public CovidCollection(Path path){
+    this.path = path;
+    this.allowedFileSuffix = Set.of(".csv");
+  }
+
+  @Override
+  public FileSegment<CovidCollection.Document> createFileSegment(Path p) throws IOException {
+    return new Segment(p);
+  }
+
+  /**
+   * A file containing a single CSV document.
+   */
+  public class Segment extends FileSegment<CovidCollection.Document> {
+    CSVParser csvParser = null;
+    private CSVRecord record = null;
+    private Iterator<CSVRecord> iterator = null; // iterator for CSV records
+
+    public Segment(Path path) throws IOException {
+      super(path);
+      bufferedReader = new BufferedReader(new InputStreamReader(
+          new FileInputStream(path.toString())));
+
+      csvParser = new CSVParser(bufferedReader, CSVFormat.DEFAULT
+        .withFirstRecordAsHeader()
+        .withIgnoreHeaderCase()
+        .withTrim());
+
+      iterator = csvParser.iterator();
+      if (iterator.hasNext()) {
+        record = iterator.next();
+      }
+    }
+
+    @Override
+    public void readNext() throws NoSuchElementException {
+      if (record == null) {
+        throw new NoSuchElementException("Record is empty");
+      } else {
+        bufferedRecord = new CovidCollection.Document(record);
+        if (iterator.hasNext()) { // if CSV contains more lines, we parse the next record
+          record = iterator.next();
+        } else {
+          atEOF = true; // there is no more JSON object in the bufferedReader
+        }
+      }
+    }
+
+    @Override
+    public void close() {
+      super.close();
+      if (csvParser != null) {
+        try {
+          csvParser.close();
+        } catch (IOException e) {
+          // do nothing
+        }
+      }
+    }
+  }
+
+  /**
+   * A document in a CORD-19 collection.
+   */
+  public class Document implements SourceDocument {
+    private String id;
+    private String contents;
+    private CSVRecord record;
+
+    public Document(CSVRecord record) {
+      id = Long.toString(record.getRecordNumber());
+      contents = record.toString();
+      this.record = record;
+
+      // index full text into raw contents
+      if (record.get("has_full_text").contains("True")) {
+        String[] hashes = record.get("sha").split(";");
+        String fullTextPath = "/" + record.get("full_text_file") + "/" + hashes[hashes.length - 1].strip() + ".json";
+        try {
+          String fullTextJson = new String(Files.readAllBytes(
+            Paths.get(CovidCollection.this.path.toString() + fullTextPath)));
+          contents += "\n " + fullTextJson;
```

Comment:
Might be good to replace all instances of "\n" with " " in the CSV record string so that this is the first newline